### ISSUE TITLE
grid_search() ignores the eval_metric set in the model, and uses loss instead to select the best hyperparameters

### ISSUE THREAD

**Issue by fantauzzi** [catboost/catboost#1457]
Problem: In spite of setting an AUC eval metric in the classifier, its grid search still selects the best hyper-parameters based on the model loss, not the eval metric I would like to use. Here a code snippet where I instantiate the classifier and run the grid search
```
clf = CatBoostClassifier(iterations=iterations,
                         eval_metric='AUC:hints=skip_train~false',
                         cat_features=cat_features,
                         random_state=seed)

grid_search_results = clf.grid_search(param_grid=param_grid,
                                      X=dev_pool,
                                      search_by_train_test_split=True,
                                      calc_cv_statistics=False,
                                      partition_random_seed=seed,
                                      verbose=True)

```

catboost version: 0.24.2
Operating System: Ubuntu 20.04.1 LTS
CPU:Intel Core™ i7-2700K CPU @ 3.50GHz × 8 
GPU:GeForce GTX 1080

**Comment by nikitxskv** [comment]
Hi @fantauzzi !

I tried to reproduce your problem but I couldn't.
For case with eval metric:
```
catboost.CatBoostClassifier(
    iterations=10,
    eval_metric='AUC:hints=skip_train~false',
).grid_search(
    param_grid={'learning_rate':[0.1, 0.01]},
    X=train_pool,
    search_by_train_test_split=True,
    calc_cv_statistics=True,
    verbose=True
);
```
i get output like this:
```
bestTest = 0.9890625
bestIteration = 3

0:	loss: 0.9890625	best: 0.9890625 (0)	total: 169ms	remaining: 169ms

bestTest = 0.990703125
bestIteration = 4

1:	loss: 0.9907031	best: 0.9907031 (1)	total: 257ms	remaining: 0us
Estimating final quality...
```
But for case without eval_metric:
```
catboost.CatBoostClassifier(
    iterations=10,
#     eval_metric='AUC:hints=skip_train~false',
).grid_search(
    param_grid={'learning_rate':[0.1, 0.01]},
    X=train_pool,
    search_by_train_test_split=True,
    calc_cv_statistics=True,
    verbose=True
);
```
i get output like this:
```
bestTest = 0.1268483344
bestIteration = 9

0:	loss: 0.1268483	best: 0.1268483 (0)	total: 256ms	remaining: 256ms

bestTest = 0.5418040226
bestIteration = 9

1:	loss: 0.5418040	best: 0.1268483 (0)	total: 315ms	remaining: 0us
Estimating final quality...
```

So, we have different metric results..
Why do you think that `eval_metric` parameter is not taken into account in your case?

**Comment by fantauzzi** [comment]
I have tried again and could not reproduce the issue (in spite of having set the random seed to the same value as before). Most likely, I misanderstood what was happening because the grid seach, in the output to console, kept reporting a "loss" even after I had selected AUC as the eval metric. That loss must be the training loss, I guess, not the eval metric.
